src/OpenTelemetry.Instrumentation.Process/ProcessMetrics.cs

```diff
@@ -24,38 +26,91 @@ internal class ProcessMetrics
             "process.memory.virtual",
-            () =>
-            {
-                CurrentProcess.Refresh();
-                return CurrentProcess.VirtualMemorySize64;
-            },
+            () => values.GetVirtualMemoryUsage(),
             unit: "By",
             description: "The amount of committed virtual memory.");
+
+        if (options.CpuStatesEnabled == false)
+        {
+            MeterInstance.CreateObservableCounter(
+            $"process.cpu.time",
+            () => values.GetProcessorCpuTime(),
+            unit: "s",
+            description: "Total CPU seconds.");
+        }
+        else
+        {
+            MeterInstance.CreateObservableCounter(
+            $"process.cpu.time",
+            () => values.GetProcessorCpuTimeWithBreakdown(),
+            unit: "s",
+            description: "Total CPU seconds broken down by different states.");
+        }
     }
 
-    /// <summary>
-    /// Initializes a new instance of the <see cref="ProcessMetrics"/> class.
-    /// </summary>
-    /// <param name="options">The options to define the metrics.</param>
-    public ProcessMetrics(ProcessInstrumentationOptions options)
+    private class InstrumentsValues
     {
+        private readonly Diagnostics.Process currentProcess = Diagnostics.Process.GetCurrentProcess();
+
+        public InstrumentsValues()
+        {
+            this.currentProcess.Refresh();
+        }
+
+        private enum CpuState
+        {
+            System,
+            User,
+            Wait,
+        }
+
+        public long GetMemoryUsage()
+        {
+            return this.currentProcess.WorkingSet64;
+        }
+
+        public long GetVirtualMemoryUsage()
+        {
+            return this.currentProcess.VirtualMemorySize64;
+        }
+
+        public long GetProcessorCpuTime()
+        {
+            return this.currentProcess.TotalProcessorTime.Seconds;
+        }
+
+        // See spec:
+        // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/process-metrics.md#metric-instruments
+        public Measurement<long>[] GetProcessorCpuTimeWithBreakdown()
+        {
+            Measurement<long>[] measurements = new Measurement<long>[Enum.GetNames(typeof(CpuState)).Length];
+            var priviledgedCpuTime = this.currentProcess.PrivilegedProcessorTime.Seconds;
+            var userCpuTime = this.currentProcess.UserProcessorTime.Seconds;
+
+            measurements[(int)CpuState.System] = new(priviledgedCpuTime, new KeyValuePair<string, object>("state", CpuState.System.ToString()));
+            measurements[(int)CpuState.User] = new(userCpuTime, new KeyValuePair<string, object>("state", CpuState.User.ToString()));
+            measurements[(int)CpuState.Wait] = new(this.currentProcess.TotalProcessorTime.Seconds - priviledgedCpuTime - userCpuTime, new KeyValuePair<string, object>("state", CpuState.Wait.ToString()));
```

Comment:
Will this give a non-zero value?

Reply:
The browser does not show me a very well-defined highlight for this comment. May I ask which part do you refer to?

Reply:
What does `TotalProcessorTime - PriviledgedCpuTime - UserCpuTime` actually mean?

Reply:
If you read this https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.totalprocessortime?view=net-6.0#property-value:

`TotalProcessorTime - PriviledgedCpuTime - UserCpuTim` is always going to give zero, unless .NET or the underlying native API has bugs.

![image](https://user-images.githubusercontent.com/17327289/187005778-171310dc-3930-4794-88bf-df08198a624d.png)